packages/vm/src/runTx.ts

```diff
@@ -339,6 +340,9 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
         throw new Error(msg)
       }
     }
+  } else {
+    fromAccount.balance += cost - balance
```

Comment:
same here, comment with an explanation

```suggestion
    // when skipBalance, ensure the caller has enough balance for the cost
    fromAccount.balance += cost - balance
```

Reply:
it's also kind of hard to follow this else condition, i suggest we change the if without the negation and do these two lines first, then else do the checks, that might be easier to parse mentally